### ISSUE TITLE
bad result in real wold environment

### ISSUE THREAD

**Issue by JasperChennn** [thu-ml/Motus#8]
When I run Real-World Inference (No Environment), I get bad results.

<img width="2880" height="384" alt="Image" src="https://github.com/user-attachments/assets/3049726d-1aad-436d-8b63-dcda2c1235c8" />

**Comment by JasperChennn** [comment]
And could you share your input image with three views? I don't have a suitable case for inference. Thanks!

**Comment by OrangeSodahub** [comment]
@HongzheBi Sorry, what I mean is the inference of Motus wan model proposed in this issue. So the Motus Stage1 (wan2.2 5B) model can be only used for three-view-concatenated inputs? What if I have only one head view?

**Comment by Alschain** [comment]
I met the similar problem while concating three-view images. Furthur expr, I screenshot the head image, left wrist image and right wrist from the paper, concat like the quoted first-frame image,  with following instruction
> ![Image](https://github.com/user-attachments/assets/63f7a1c4-55cb-48cf-a91c-9175ba8da0fb)

```
python inference/real_world/Motus/inference_example.py \
  --model_config inference/real_world/Motus/utils/ac_one.yaml \
  --ckpt_dir pretrained_models/Motus \
  --wan_path pretrained_models \
  --image examples/sample_frame.png \
  --instruction "Get water from water dispenser" \
  --use_t5 \
  --output examples/output.png
```

For `ac_one.yaml`, just modify the model path; for action expert, just ignoring the unmatched pos_embedding loading and the missing registers & state encoder. But the predicted results appear poor, similar to the issue image.

Is there something wrong when using related methods/models?

**Comment by waaadreamer** [comment]
The provided checkpoints are from Stage 2 (Latent Action) training. They have zero exposure to real-world environments or physics and represent latent action only. You must collect domain-specific data and perform Stage 3 SFT to align the latent actions with real-world execution.

**Comment by Alschain** [comment]
> The provided checkpoints are from Stage 2 (Latent Action) training. They have zero exposure to real-world environments or physics and represent latent action only. You must collect domain-specific data and perform Stage 3 SFT to align the latent actions with real-world execution.

Gotcha and thanks.

So is that mean in Motus pretraining stage, due to learning better latent action with large data, language-video data is a small proportion, and this makes the model in pretrained stage not good enough for directly predicting future (video) states with observation and instruction? While after SFT, with specific data training, the finetuned model could handle such things well?

**Comment by HongzheBi** [comment]
> I met the similar problem while concating three-view images. Furthur expr, I screenshot the head image, left wrist image and right wrist from the paper, concat like the quoted first-frame image, with following instruction
> 
> > ![Image](https://github.com/user-attachments/assets/63f7a1c4-55cb-48cf-a91c-9175ba8da0fb)
> 
> ```
> python inference/real_world/Motus/inference_example.py \
>   --model_config inference/real_world/Motus/utils/ac_one.yaml \
>   --ckpt_dir pretrained_models/Motus \
>   --wan_path pretrained_models \
>   --image examples/sample_frame.png \
>   --instruction "Get water from water dispenser" \
>   --use_t5 \
>   --output examples/output.png
> ```
> 
> For `ac_one.yaml`, just modify the model path; for action expert, just ignoring the unmatched pos_embedding loading and the missing registers & state encoder. But the predicted results appear poor, similar to the issue image.
> 
> Is there something wrong when using related methods/models?


This problem is actually quite complex.

1. First, you're using stage2 weights here, and we generally don't directly use stage2 weights for inference. If you must use stage2 weights for inference, the yaml config should write `latent_action.yaml` (which is equivalent to latent action being a robot body), and the generation effect would probably be better. Because if you fill in `ac-one` here, the action encoder and decoder weights in the action expert part don't match, so as you mentioned, you directly ignore them, i.e., don't load them. However, our inference code performs joint action-video denoising, which means these weights become completely new. Unless you modify the inference function to perform separate denoising of the video marginal distribution, i.e., keep the action part as pure noise all the time, it might improve. But overall, we still don't recommend directly using stage2 models for inference—we haven't even tried it.

2. Additionally, video models and world models will definitely perform better in-domain, so we recommend doing some SFT fine-tuning.

**Comment by Alschain** [comment]
Thanks for your reply, And in my futhur test, with the robotwin sft model checkpoint, the predicted video **is quite better for robotwin data and real-world data.**

While using Stage 2 pretraining checkpoint with `latent_action.yaml`, no matter keeping action part as pure noise or not, the generated video is still in poor quality. I get something confused:

1. In **Motus Pretraining Stage**, level 2 to level 5 data is all used for pretraining, while human based egocentric data only contains head view while robot related data is three views in T-pose, is such data treated separately?
2. In my opinion, in **Motus Pretraining Stage** with video expert joint training with related loss, we could get the satisfying generated video with first frame and related instruction using this stage checkpoint, but during my testing it appears not in that way. Or is there something different from my thought?
